### DISCUSSION TITLE
Fields of figure caption

### DISCUSSION THREAD

**Discussion by Julian-Wassmann** [typst/typst#3333] [General]
To style the caption of a figure, it is possible to apply a show rule to `figure.caption`. In the [docs](https://typst.app/docs/reference/model/figure/#definitions-caption) it is claimed that the `caption` contains following fields:
- belonging to the `caption` element:
  - `pos`
  - `body`
- belonging to the parent `figure` element:
  - `kind`
  - `supplement`
  - `counter`
  - `numbering`
  - `location`

First of all, the position field is named `position`, not `pos`. I guess that is a typo in the docs. Secondly, the `caption` does contain the field `seperator`. Thirdly, it does not contain the field `location`. You can see this in the following example:
```typst
#show figure.caption: it => [
  #it
  
  `it.location() = `#{type(it.location())}
  
  `it.fields() = `#it.fields()
]

#figure(
  rect[Hello],
  caption: [A rectangle],
)
```

While `location` is not a field, there still is the method `location()`. However, it seems to always return `none`. Why is this the case? From my understanding, the location and some other fields should be cloned from the parent `figure` element. For the figure, the location method is returning the location correctly. Of course, it is possible to work around this issue, which is not my point here.

So two things are mixed together here:
1. The documentation seems to be a little bit wrong / outdated.
2. The location method returning `none` seems like a bug to me.

What are your thoughts about this?

**Comment by laurmaedje** [comment]
The docs are outdated. The location is none because the caption would only get a location once it itself is shown (and only if it was locatable, which it isn't).
